modules/node-shared/src/main/scala/io/constellationnetwork/node/shared/infrastructure/snapshot/CurrencySnapshotCreator.scala

```diff
@@ -154,6 +157,16 @@ object CurrencySnapshotCreator {
                     .getOrElse(SortedSet.empty[RewardTransaction].pure[F])
               )
 
+          rejectedBlockEvents = NonEmptyList.fromListUnsafe(acceptanceResult.notAccepted.collect {
+            case (b, r: BlockRejectionReason) => (BlockEvent(b), Some(r))
+          })
+          _ <- currencySnapshotValidationErrorStorage.addValidationErrors(rejectedBlockEvents)
```

Comment:
`notAccepted` can be empty, so it will throw if you use `fromListUnsafe` when all the blocks are accepted. The traverse on `Option` is safer way:
```scala
_ <- NonEmptyList.from(acceptanceResult.notAccepted.collect {
     case (b, r: BlockRejectionReason) => (BlockEvent(b), Some(r))
   }).traverse(currencySnapshotValidationErrorStorage.addValidationErrors)
```